src/github.com/getlantern/flashlight/config/config.go

```diff
@@ -1,339 +1,263 @@
+	// ProxiesURLs are the chained and fronted URLs for fetching the per user
+	// proxy config.
+	ProxiesURLs = &ChainedFrontedURLs{
+		Chained: "http://config.getiantem.org/proxies.yaml.gz",
+		Fronted: "http://d2wi0vwulmtn99.cloudfront.net/proxies.yaml.gz",
 	}
 
-	m = &yamlconf.Manager{
-		FilePath: configPath,
-
-		EmptyConfig: func() yamlconf.Config {
-			return &Config{configDir: configDir}
-		},
-		PerSessionSetup: func(ycfg yamlconf.Config) error {
-			cfg := ycfg.(*Config)
-			return cfg.applyFlags(flags)
-		},
-		// Obfuscate on-disk contents of YAML file
-		Obfuscate: flags["readableconfig"] == nil || !flags["readableconfig"].(bool),
+	// ProxiesStagingURLs are the chained and fronted URLs for fetching the per user
+	// proxy config in a staging environment.
+	ProxiesStagingURLs = &ChainedFrontedURLs{
+		Chained: "http://config-staging.getiantem.org/proxies.yaml.gz",
+		Fronted: "http://d33pfmbpauhmvd.cloudfront.net/proxies.yaml.gz",
 	}
+)
 
-	if stickyConfig {
-		log.Debug("stickyconfig set, using config from disk as-is")
-	} else {
-		m.ValidateConfig = validateConfig
-		m.DefaultConfig = MakeInitialConfig
-		m.CustomPoll = func(ycfg yamlconf.Config) (mutate func(yamlconf.Config) error, waitTime time.Duration, err error) {
-			return fetcher.pollForConfig(ycfg)
-		}
-	}
-	initial, err := m.Init()
+// Config is an interface for getting proxy data saved locally, embedded
+// in the binary, or fetched over the network.
+type Config interface {
 
-	var cfg *Config
-	if err != nil {
-		log.Errorf("Error initializing config: %v", err)
-	} else {
-		cfg = initial.(*Config)
-	}
-	log.Debug("Returning config")
-	return cfg, err
+	// Saved returns a yaml config from disk.
+	saved() (interface{}, error)
+
+	// Embedded retrieves a yaml config embedded in the binary.
+	embedded([]byte, string) (interface{}, error)
+
+	// Poll polls for new configs from a remote server and saves them to disk for
+	// future runs.
+	poll(UserConfig, chan interface{}, *ChainedFrontedURLs, time.Duration)
 }
 
-// Run runs the configuration system.
-func Run(updateHandler func(updated *Config)) error {
-	for {
-		next := m.Next()
-		nextCfg := next.(*Config)
-		updateHandler(nextCfg)
-	}
+type config struct {
+	filePath  string
+	obfuscate bool
+	saveChan  chan interface{}
+	factory   func() interface{}
 }
 
-// Update updates the configuration using the given mutator function.
-func Update(mutate func(cfg *Config) error) error {
-	return m.Update(func(ycfg yamlconf.Config) error {
-		return mutate(ycfg.(*Config))
-	})
+// ChainedFrontedURLs contains a chained and a fronted URL for fetching a config.
+type ChainedFrontedURLs struct {
+	Chained string
+	Fronted string
 }
 
-func inConfigDir(configDir string, filename string) (string, string, error) {
-	cdir := configDir
+// PipeConfig creates a new config pipeline for reading a specified type of
+// config onto a channel for processing by a dispatch function.
+func PipeConfig(configDir string, obfuscate bool,
```

Comment:
Hmm, it does nothing better than the current method. We can do a small change here, though: merge the parameters to a single `Options` struct, so from the caller side, we know what's the purpose of each parameter, and how they are strongly correlated.